src/OpenSpaceToolkit/Astrodynamics/Estimator/ODLeastSquaresSolver.cpp

```diff
@@ -0,0 +1,178 @@
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/StateBuilder.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace estimator
+{
+
+using ostk::astrodynamics::trajectory::Orbit;
+using ostk::astrodynamics::trajectory::orbit::model::Propagated;
+using ostk::astrodynamics::trajectory::StateBuilder;
+
+const Shared<const Frame> ODLeastSquaresSolver::EstimationFrameSPtr = Frame::GCRF();
+
+ODLeastSquaresSolver::Analysis::Analysis(const State& aDeterminedState, const LeastSquaresSolver::Analysis& anAnalysis)
+    : determinedState_(aDeterminedState),
+      solverAnalysis_(anAnalysis)
+{
+}
+
+std::ostream& operator<<(std::ostream& anOutputStream, const ODLeastSquaresSolver::Analysis& anAnalysis)
+{
+    anAnalysis.print(anOutputStream);
+
+    return anOutputStream;
+}
+
+void ODLeastSquaresSolver::Analysis::print(std::ostream& anOutputStream) const
+{
+    ostk::core::utils::Print::Header(anOutputStream, "Orbit Determination Solver Analysis");
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Determined State");
+    determinedState_.print(anOutputStream, false);
+
+    ostk::core::utils::Print::Separator(anOutputStream, "Analysis");
+    solverAnalysis_.print(anOutputStream);
+
+    ostk::core::utils::Print::Footer(anOutputStream);
+}
+
+const State& ODLeastSquaresSolver::Analysis::getDeterminedState() const
+{
+    return determinedState_;
+}
+
+const LeastSquaresSolver::Analysis& ODLeastSquaresSolver::Analysis::getSolverAnalysis() const
+{
+    return solverAnalysis_;
+}
+
+ODLeastSquaresSolver::ODLeastSquaresSolver(
+    const Environment& anEnvironment, const NumericalSolver& aNumericalSolver, const LeastSquaresSolver& aSolver
+)
+    : environment_(anEnvironment),
+      propagator_(Propagator::FromEnvironment(aNumericalSolver, anEnvironment)),
+      solver_(aSolver)
+{
+    if (!environment_.hasCentralCelestialObject())
+    {
+        throw ostk::core::error::runtime::Undefined("Central celestial object");
+    }
+}
+
+const Environment& ODLeastSquaresSolver::accessEnvironment() const
+{
+    return environment_;
+}
+
+const Propagator& ODLeastSquaresSolver::accessPropagator() const
+{
+    return propagator_;
+}
+
+const LeastSquaresSolver& ODLeastSquaresSolver::accessSolver() const
+{
+    return solver_;
+}
+
+ODLeastSquaresSolver::Analysis ODLeastSquaresSolver::estimateState(
+    const State& anInitialGuessState,
+    const Array<State>& aReferenceStateArray,
+    const Array<Shared<const CoordinateSubset>>& anEstimationCoordinateSubsets,
+    const std::unordered_map<CoordinateSubset, VectorXd>& anInitialGuessSigmas,
+    const std::unordered_map<CoordinateSubset, VectorXd>& aReferenceStateSigmas
+) const
+{
+    // Convert everything to an inertial frame for estimation
+
+    const State initialGuessState = anInitialGuessState.inFrame(EstimationFrameSPtr);
+
+    const Array<State> referenceStates = aReferenceStateArray.map<State>(
```

Comment:
Same here, I think this would make it more readable:

```suggestion
    const Array<State> referenceStatesInFrame = aReferenceStateArray.map<State>(
```